File: Java/475.java
__________________________________________________________________________________________________
sample 7 ms submission
class Solution {
    public int findRadius(int[] houses, int[] heaters) {
        Arrays.sort(houses);
        Arrays.sort(heaters);
        int i = 0, res = 0;
        for(int h: houses) {
            while(i < heaters.length - 1 && heaters[i] + heaters[i + 1] <= h * 2) i++;
            res = Math.max(res, Math.abs(heaters[i] - h));
        }
        return res;
    }
}
__________________________________________________________________________________________________
sample 8 ms submission
class Solution {
    public int findRadius(int[] houses, int[] heaters) {
        if(houses == null || heaters == null || houses.length == 0 || heaters.length == 0)
        		return -1;
        Arrays.sort(houses);
        Arrays.sort(heaters);
        
        int max = 0,i = 0;
        
        for(int house : houses) {
        		while(i < heaters.length - 1 && heaters[i] + heaters[i + 1] <= house * 2)
        			i++;
        			max = Math.max(max, Math.abs(house - heaters[i]));
        		}
        
        return max;
    }
}
__________________________________________________________________________________________________
sample 40136 kb submission
public class Solution {
    public int findRadius(int[] houses, int[] heaters) {
        Arrays.sort(houses);
        Arrays.sort(heaters);
        
        int i = 0, res = 0;
        for (int house : houses) {
            while (i < heaters.length - 1 && heaters[i] + heaters[i + 1] <= house * 2) {
                i++;
            }
            res = Math.max(res, Math.abs(heaters[i] - house));
        }
        
        return res;
    }
}

File: Python3/475.py
__________________________________________________________________________________________________
sample 276 ms submission
class Solution:
    def findRadius(self, houses: List[int], heaters: List[int]) -> int:
        heaters.sort()
        heaters1=set(heaters)
        rslt=0
        for h in houses:
            if h not in heaters1:   
                pos=bisect.bisect(heaters,h)
                if pos==0:
                    rslt=max(rslt,heaters[0]-h)
                elif pos==len(heaters):
                    rslt=max(rslt,h-heaters[-1])
                else:                
                    rslt=max(rslt,min(h-heaters[pos-1],heaters[pos]-h))
        return rslt 
__________________________________________________________________________________________________
sample 15640 kb submission
import bisect
class Solution:
    def findRadius(self, houses: 'List[int]', heaters: 'List[int]') -> 'int':
        houses.sort()
        heaters.sort()
        ans = 0
        for i in houses:
            temp = bisect.bisect_left(heaters, i)
            if temp == 0 :
                ans = max(ans,abs(heaters[temp] - i))
            elif temp == len(heaters):
                ans = max(ans,abs(heaters[temp - 1] - i))
            else:
                ans = max(ans,min(abs(heaters[temp] - i), abs(heaters[temp - 1] - i)))
        return ans
__________________________________________________________________________________________________
